src/routes/Profile/Competences/GroupBySource.tsx

```diff
@@ -0,0 +1,60 @@
+import { Accordion, ConfirmDialog, Tag } from '@jod/design-system';
+import { t } from 'i18next';
+import React from 'react';
+import type { GroupByProps } from './constants';
+
+export const GroupBySource = ({
+  filter,
+  filterKeys,
+  locale,
+  osaamiset,
+  deleteOsaaminen,
+  isOsaaminenVisible,
+}: GroupByProps) => {
+  return (
+    <>
+      <h2 className="mb-6 text-heading-3">Osaamiseni lähteiden mukaan</h2>
+      {filterKeys.map((key) => {
+        return (
+          Array.isArray(filter[key]) &&
+          filter[key]!.some((x) => x.checked) && (
```

Comment:
Just little nitpicking, maybe using something more describing than `x`, would it be e.g `filter` and maybe because of this to have also filter renamed.

So in the end it would be a little more readable perhaps:
```tsx
filters[key]!.some((filter) => filter.checked)
```

Reply:
This seems to be changed now 👍

Reply:
Renamed. Also renamed the `filter` object to `filters`, as it actually contains all filters.

While I was at it, I fixed two other issues:
* In alphabet grouping it would show skills from a filtered toimenkuva (forgot to add a filter check).
* Unchecked filters would get "rechecked" after deletion of a skill.